File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& l) 
    {
        int i;
        for(i=0;i<l.size()-1;i++)
        {
            while(l[i]-1!=i && l[i]!=l[l[i]-1])
            {
                swap(l[i],l[l[i]-1]);
            }
            if(l[i]==l[l[i]-1] && l[i]-1!=i)
            {
                return(l[i]);
            }
        }
        return(l[l.size()-1]);
  //       int slow = nums[0];
  // int fast = nums[0];
  // do {
  //   slow = nums[slow];
  //   fast = nums[nums[fast]];
  // } while (slow != fast);
  // fast = nums[0];
  // while (slow != fast) {
  //   slow = nums[slow];
  //   fast = nums[fast];
  // }
  // return slow;
    }
};

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.py
class Solution:
    def findDuplicate(self, l: List[int]) -> int:
        for i in range(len(l)-1):
            while(l[i]-1!=i and l[i]!=l[l[i]-1]):
                l[l[i]-1],l[i]=l[i],l[l[i]-1]
                # x=l[i]
                # l[i]=l[l[i]-1]
                # l[x-1]=x
            if(l[i]==l[l[i]-1] and l[i]-1!=i):
                return(l[i])
        return(l[len(l)-1])